### PR TITLE
[IMPROVEMENT] added build checks for builddebug, autoconf, cmake

### PR DESCRIPTION
**Pull request by satyammittal**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/canihavesomecoffee/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [x] I am an active contributor to the project.

---

{pull request content here}
#55 #18 
added build checks for builddebug, autoconf, cmake

### CODE REVIEW

**Comment by saurabhshri** [comment]
One of the main reasons to include CMake test is to enable compilation with non Visual Studio IDEs like CLion which make use of CMakeLists. Thus, it is equally important to check if cmake is success in Windows as well.

**Comment by satyammittal** [comment]
I am working on it. This pr is only for Linux.

File: install/ci-vm/ci-linux/ci/runCI [hunk 1]
```diff
@@ -49,6 +49,52 @@ function executeCommand {
...
 	fi
 }
+# Check whether cmake is a success or not
+function cmakeCheck {
+	executeCommand mkdir build
+	executeCommand cd build
+	executeCommand cmake ../src/
+	executeCommand make
+	if [ -x "ccextractor" ]; then
+		postStatus "cmake building" "Successful build"
```
Comment (canihavesomecoffee) [inline install/ci-vm/ci-linux/ci/runCI]: This will cause a crash in the platform if I merge this.

File: install/ci-vm/ci-linux/ci/runCI [hunk 2]
```diff
@@ -49,6 +49,52 @@ function executeCommand {
...
+	fi
+	executeCommand cd ..
+	executeCommand rm -rf build
+}
+# Check whether builddebug is a success or not
+function builddebugCheck {
+	executeCommand cd linux
+	./builddebug >> "${logFile}" 2>&1
+	if [ -x "ccextractor" ]; then
+		postStatus "builddebug building" "Successful build"
```
Comment (canihavesomecoffee) [inline install/ci-vm/ci-linux/ci/runCI]: This will cause a crash in the platform if I merge this.

File: install/ci-vm/ci-linux/ci/runCI [hunk 3]
```diff
@@ -49,6 +49,52 @@ function executeCommand {
...
+}
+# Check whether autoconf is a success or not
+function autoconfCheck {
+	executeCommand cd linux
+	./autogen.sh >> "${logFile}" 2>&1
+	if [ -x "configure" ]; then
+		./configure >> "${logFile}" 2>&1
+		executeCommand make
+		if [ -x "ccextractor" ]; then
+			postStatus "autoconf building" "Successful build"
```
Comment (canihavesomecoffee) [inline install/ci-vm/ci-linux/ci/runCI]: This will cause a crash in the platform if I merge this.

File: install/ci-vm/ci-linux/ci/runCI [hunk 4]
```diff
@@ -49,6 +49,52 @@ function executeCommand {
 		haltAndCatchFire "" # No message needed as we post before anyway
 	fi
 }
+# Check whether cmake is a success or not
+function cmakeCheck {
+	executeCommand mkdir build
+	executeCommand cd build
+	executeCommand cmake ../src/
+	executeCommand make
```
Comment (canihavesomecoffee) [inline install/ci-vm/ci-linux/ci/runCI]: What if make fails?

File: install/ci-vm/ci-linux/ci/runCI [hunk 5]
```diff
@@ -49,6 +49,52 @@ function executeCommand {
...
+function builddebugCheck {
+	executeCommand cd linux
+	./builddebug >> "${logFile}" 2>&1
+	if [ -x "ccextractor" ]; then
+		postStatus "builddebug building" "Successful build"
+		executeCommand rm ccextractor
+	else
+		postStatus "builddebug building" "Failed to build"
+	fi
+	./cleanup
```
Comment (canihavesomecoffee) [inline install/ci-vm/ci-linux/ci/runCI]: Why run a make-related cleanup if you build with builddebug?

File: install/ci-vm/ci-linux/ci/runCI [hunk 6]
```diff
@@ -57,6 +103,15 @@ postStatus "preparation" "Copy code to local folder"
 executeCommand cp -r ${srcDir}/* ${dstDir}
 executeCommand cd ${dstDir}
 
+postStatus "cmake building" "Compiling CCExtractor using cmake"
```
Comment (canihavesomecoffee) [inline install/ci-vm/ci-linux/ci/runCI]: What if any of these fails? We would like to run the tests with the "regular" built version regardless of build failures on one of these three...

Comment (canihavesomecoffee) [inline install/ci-vm/ci-linux/ci/runCI]: Please test it out on the server you have available...

**Pull request merged**